registry/logging.go

```diff
@@ -0,0 +1,97 @@
+package registry
+
+import (
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+)
+
+var json = jsoniter.ConfigCompatibleWithStandardLibrary
+
+// jsonLogEntry represents a log entry in JSON format.
+type jsonLogEntry struct {
+	Timestamp time.Time `json:"timestamp"`
+	Method    string    `json:"method"`
+	Path      string    `json:"path"`
+	Status    int       `json:"status"`
+	Size      int       `json:"size"`
+	Referer   string    `json:"referer"`
+	UserAgent string    `json:"user_agent"`
+}
+
+// jsonLogFormatterParams holds the parameters required for JSON logging.
+type jsonLogFormatterParams struct {
+	Request   *http.Request
+	URL       *url.URL
+	Timestamp time.Time
+	Status    int
+	Size      int
+}
+
+type responseLogger struct {
+	w      http.ResponseWriter
+	status int
+	size   int
+}
+
+func (l *responseLogger) Write(p []byte) (int, error) {
+	size, err := l.w.Write(p)
+	l.size += size
+	return size, err
+}
+
+func (l *responseLogger) Status() int {
+	if l.status == 0 {
+		return http.StatusOK
+	}
+	return l.status
+}
+
+func (l *responseLogger) Size() int {
+	return l.size
+}
+
+func makeLogger(w http.ResponseWriter) *responseLogger {
+	return &responseLogger{w: w, status: http.StatusOK}
```

Comment:
This middleware is incomplete. It does not capture the actual response status code. Unfortunately you will have to wrap the `http.ResponseWriter` in order to capture the status code parameter when the `WriteHeader` method is called.